### PR TITLE
buy crypto

### PR DESCRIPTION
**Pull request by cammellos**
Implements buy on crypto ramp.

status: ready

### CODE REVIEW

**Comment by errorists** [comment]
UI:

- [x] Needs colour mapping for Dark Mode

IMG_0110

- [x] the `new` badge is cropped vertically
- [ ] the modal with the list of services isn't presented in a modal dialogue
- [ ] the modal with the list of services, the scrollable area should include the header. only the 56 point high top bar has fixed position.
- [ ] webview should be presented modally

File: src/status_im/events.cljs [hunk 1]
```diff
@@ -1291,3 +1294,25 @@
                 (reset-current-profile-chat % (get-in % [:db :contacts/identity]))
 
                 nil))))
+
+(defn on-ramp<-rpc [on-ramp]
+  (clojure.set/rename-keys on-ramp {:logoUrl :logo-url
+                                    :siteUrl :site-url}))
+
+(handlers/register-handler-fx
```
Comment (flexsurfer) [inline src/status_im/events.cljs:1302]: events.cljs and register-handler-fx are legacy, we use model.cljs and fx/defn instead

File: src/status_im/ui/screens/wallet/account/views.cljs [hunk 2]
```diff
@@ -37,7 +38,7 @@
   [react/touchable-highlight {:on-press handler :style {:flex 1}}
    [react/view {:flex 1 :align-items :center :justify-content :center}
     [react/view {:flex-direction :row :align-items :center}
-     [icons/icon icon {:color color}]
+     [vector-icons/icon icon {:color color}]
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/account/views.cljs]: why renamed ?

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/ui/screens/wallet/account/views.cljs (hunk 2 above)
Comment (cammellos) [reply src/status_im/ui/screens/wallet/account/views.cljs]: Oh, we shuffled things around, and imported and remamed etc, we'll change it back ;)

**Comment by Serhy** [comment]
App crashes when I tap on any service from Buy crypto list. Reproduced on Android 11, however it's all good on Android 8.

This is logcat snippet before tap on some service:

```
02-05 16:40:48.909 23950 23950 I ViewRootImpl@fe3f1cd[MainActivity]: ViewPostIme pointer 0
02-05 16:40:48.965 23950 23950 I ViewRootImpl@fe3f1cd[MainActivity]: ViewPostIme pointer 1
02-05 16:40:48.976 23950 27123 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :navigate-to
02-05 16:40:48.978 23950 27123 D ReactNativeJS: DEBUG [status-im.navigation:10] - :navigate-to :buy-crypto-website {:name "Ramp", :description "Global crypto to fiat flow", :fees "1.5%", :region "Global", :hostname "ramp.network", :logo-url "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAARgAAAEYCAMAAACwUBm+AAAABGdBTUEAALGPC/xhBQAAAAFzUkdCAK7OHOkAAAAgY0hSTQAAeiYAAICEAAD6AAAAgOgAAHUwAADqYAAAOpgAABdwnLpRPAAAAKtQTFRF////8fv3kN+5WM+WL8N7Ib9zS8uNgtuw4/fu0eThhLauOIl6GndmCm5cR5KFlL+44e3rdNen8Pb1da2kPceEKYBw1fPlGXdm0eTg8fv2ZqSZL8N8x+/cwtvW4O3rrOfLR5KEo8nCq+fLhbetnuPCddenOIl74/fthbats9HMLsN8uevUk7+3nePChLeuZtOfV5uPZqWZ7/b1lMC3o8jCKIBxN4l7SsuNda2jc0pF4QAABfxJREFUeJzt3Wt32kYQh/ENIMBxlBhKoMQOTmKn9Jbek/b7f7Ianzq6oJV2Z0a7Mz3/52185hz9IoQEK+EcQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQs2eT6ayYzRdLwZkXzy9flC9evroSnJm6yaz4r5UUzfqb8qnnQiOT92xeVG1kZK5el1WXa5GZqWu4CMk0XIzKtFxEZFouJmWWbRcBmTOXsnxt7RC83Jy5sGU6XMzJdLowZTpdjMl4XFgyHhdTMl4XhozXxZBMjwtZpsfFjEyvC1Gm18WIzIALSWbAxYTMoAtBZtDFgEyAS7RMgIt6mSCXSJkgF+UygS5RMoEuqmWCXSJkgl0Uy0S4BMtEuKiViXIJlIlyUSoT6RIkE+miUmYb6xIgE+2iUGY7G4aIlSG4lOXu20RbHBbJZUCG5KJMhujyILP3ziS6qJIhuxTFxDv0DdFFkQzDpbj2Tt2RYbTIcFyKYuuZekF3USLDcyluPGPfcmA0yDBdCt/R98CCyS9zy2Mp5t7J73gy5fuECudxXYoP3tG811JmGbaL/93auTu7MqO6GJYZ2cWszOguRmXkXbY3Z2d7BmWkXZaL0wnRrL1Wz5zMvbDL7dN54nXrDZwtk3YJ40TapfZPlmXGdLEsM66LXZmxXazKjO9iU0bapfv9rfVHBmSkXXzzrMmkcrEmk87FlkxKF0syaV3syKR2sSKT3sWGTA4XCzJ5XPTLLDK5aJf5mM1FXubN/8RFs0xeF70yuV20yuR30SmjwUWjDNvlo4TLmQx9hZ6QDNultS6Tfp7YlFl/l1Xm7DEE8R2FXNoyzBVXJevxBwIuza3hXVc0Z32fT0bAZdPYYbjXWw2ZNWO9K09GwKVYSbq0ZPi7DFFGwKWxipfv0pRhL9J76CXBRWI7NrV57PURj93XJlJvNqgXf629lNiMlfC8oriurZ9hn8s8tIt+MbE/gDlV+/9dScx7aFqNfCUAE7/LTCW2oloPv5cYd+q6OmH8QQIm+ijzo8RWVMfeG4lxrZms+zCe2sXCiGxEdUCQg6n2Qv7J76nMe8xWYtxjFUyePUbkGPPT13F7EehT1TFG4kQm/hgjcRpTf1cSgS4aZwAi70o/x8KI/BcvqnlHoV2mdvElcR5THmJh3CeBrdgIz5M/8/0l2sW5XwW2o35jn/S1ksSx9zeCi9sLnKw2LoeFr64FXkm//0GBkTiNv258sCn6eQz5hvWaC41FREbzJ3h0FwmZ5t0kcp/5XmV1EZCZj/QtAfuFxHMRkFk152n5XonrIiAzxjeR7AMM30WHjEYXDTI6XfKtvxvLhXIdoFFGr0teGc0uOWV0u6hZ56vORcnKcIUuKu4lUOmi4O4TpS7Z71dS65L5DjfFLlnviVTtkvEuWuUu2e67Vu+SSua++UcGXORlPnT8ya1BlxQyNl3Gl7HqIvA9dK+MtMufyVy6jwtSMtIuaZ9rJi7ztEpkZttFXua4enwO3qJ5t4o9F3mZzicnGnQZQ+Ysky4JZIy6jC5j1mVkGcMuzt1w12L6nr7v3F9Mlt3bhA7nbZkyU+9k5m2yu4uECl1xZXy/l7M27sKWGemXLPK7cGXG+e0TDS48mXF+LUeHC0tm5R1Kf7dW48KQ2Ry9M9ef7buQZXpcnDvQZFS5OHeM/zXEAReizOf4+2zGjSAz4EKSUedCkBl0IcgodImWCXCJllHpEikT5BIpo9QlSibQJUpGrUuETLBLhIxil2CZCJdgGdUugTJRLoEyyl2CZCJdgmTUuwTIRLsEyBhwGZQhuAzKmHAZkCG5DMgYcemVIbr0yphx6ZEhu/TIGHLxyjBcvDKmXDwyLBePjDGXThmmS6fMF2suzu3/lnbpkPlCfJxH1loyAi5nMiZdWjIiLi0Zoy6uviB4sR/+66AOX5/qsLsTGpmj46fprJjNJ74vYikd/rnclbt3d2Z3F4QQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIe39C9vesmAqb7TPAAAAAElFTkSuQmCC", :site-url "https://ramp.network/buy/"}
02-05 16:40:49.050 23950 23950 W unknown:WrappingUtils: Don't know how to round that drawable: com.facebook.drawee.drawable.RoundedColorDrawable@5725cbe
02-05 16:40:49.056 23950 27123 D ReactNativeJS: DEBUG [status-im.ui.screens.routing.core:56] - :on-screen-focus :buy-crypto-website
02-05 16:40:49.063 23950 27123 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :screens/on-will-focus
02-05 16:40:49.064 23950 27123 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :set
02-05 16:40:49.082 23950 27057 W cr_media: Requires BLUETOOTH permission
02-05 16:40:49.092 23950 27363 I mali_egl: eglDestroySurface() in
02-05 16:40:49.093 23950 27363 I mali_egl: eglDestroySurface() out
02-05 16:40:49.102 23950 23950 D SurfaceControl: release : Surface(name=im.status.ethereum.pr/im.status.ethereum.MainActivity$_23950)/@0x82a9043
02-05 16:40:49.102 23950 23950 I SurfaceControl: nativeRelease nativeObject s[534425206720]
02-05 16:40:49.102 23950 23950 I SurfaceControl: nativeRelease nativeObject e[534425206720]
02-05 16:40:49.103 23950 23950 D SurfaceControl: release : Surface(name=Surface(name=cc3221b InputMethod)/@0x5bc4f64 - animation-leash)/@0x9edc6ee
02-05 16:40:49.103 23950 23950 I SurfaceControl: nativeRelease nativeObject s[534425134704]
02-05 16:40:49.103 23950 23950 I SurfaceControl: nativeRelease nativeObject e[534425134704]
02-05 16:40:49.103 23950 23950 D SurfaceControl: release : Surface(name=Surface(name=e793c62 NavigationBar0)/@0xaae0de3 - animation-leash)/@0x6ee678f
02-05 16:40:49.103 23950 23950 I SurfaceControl: nativeRelease nativeObject s[534425209072]
02-05 16:40:49.103 23950 23950 I SurfaceControl: nativeRelease nativeObject e[534425209072]
02-05 16:40:49.103 23950 23950 D SurfaceControl: release : Surface(name=Surface(name=b7534b7 StatusBar)/@0x146d7a5 - animation-leash)/@0xea9da1c
02-05 16:40:49.103 23950 23950 I SurfaceControl: nativeRelease nativeObject s[534425209744]
02-05 16:40:49.103 23950 23950 I SurfaceControl: nativeRelease nativeObject e[534425209744]
02-05 16:40:49.103 23950 23950 I ViewRootImpl@fe3f1cd[MainActivity]: Relayout returned: old=(0,0,1080,2400) new=(0,0,1080,2400) req=(1080,2400)0 dur=2 res=0x1 s={true 539251277808} ch=false fn=852
02-05 16:40:49.105 23950 27363 W AudioCapabilities: Unsupported mime audio/x-ima
02-05 16:40:49.105 23950 27363 W VideoCapabilities: Unsupported mime video/mp43
02-05 16:40:49.105 23950 27363 W VideoCapabilities: Unsupported mime video/wvc1
02-05 16:40:49.105 23950 27363 W VideoCapabilities: Unsupported mime video/x-ms-wmv
02-05 16:40:49.105 23950 27363 W AudioCapabilities: Unsupported mime audio/x-ms-wma
02-05 16:40:49.105 23950 27363 W VideoCapabilities: Unsupported mime video/x-ms-wmv7
02-05 16:40:49.105 23950 27363 W VideoCapabilities: Unsupported mime video/x-ms-wmv8
02-05 16:40:49.106 23950 27363 W AudioCapabilities: Unsupported mime audio/x-ima
02-05 16:40:49.106 23950 27363 W AudioCapabilities: Unsupported mime audio/mpeg-L1
02-05 16:40:49.106 23950 27363 W AudioCapabilities: Unsupported mime audio/mpeg-L2
02-05 16:40:49.106 23950 27363 W AudioCapabilities: Unsupported mime audio/x-ms-wma
02-05 16:40:49.106 23950 27363 W VideoCapabilities: Unsupported mime video/mp43
02-05 16:40:49.106 23950 27363 W VideoCapabilities: Unsupported mime video/wvc1
02-05 16:40:49.106 23950 27363 W VideoCapabilities: Unsupported mime video/x-ms-wmv7
02-05 16:40:49.106 23950 27363 W VideoCapabilities: Unsupported mime video/x-ms-wmv8
02-05 16:40:49.108 23950 27363 W AudioCapabilities: Unsupported mime audio/x-ape
02-05 16:40:49.110 23950 27363 W VideoCapabilities: Unsupported mime video/avc-wfd
02-05 16:40:49.110 23950 27363 W VideoCapabilities: Unsupported mime video/avc-wfd
02-05 16:40:49.110 23950 27363 W VideoCapabilities: Unsupported mime video/avc-wfd
02-05 16:40:49.111 23950 27363 W VideoCapabilities: Unsupported mime video/hevc-wfd
02-05 16:40:49.111 23950 27363 W VideoCapabilities: Unsupported mime video/hevc-wfd
02-05 16:40:49.114 23950 27363 W VideoCapabilities: Unrecognized profile/level 1/32 for video/mp4v-es
02-05 16:40:49.114 23950 27363 W VideoCapabilities: Unrecognized profile/level 32768/2 for video/mp4v-es
02-05 16:40:49.114 23950 27363 W VideoCapabilities: Unrecognized profile/level 32768/64 for video/mp4v-es
--------- beginning of crash
02-05 16:40:49.194 23950 27078 F libc    : Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x20 in tid 27078 (RenderThread), pid 23950 (tus.ethereum.pr)
^C

```

Full device logcat: 
[buy_crypto_crash.txt](https://github.com/status-im/status-react/files/5933307/sdsdcscrash.txt)

File: src/status_im/ui/screens/wallet/account/views.cljs (hunk 2 above)
Comment (Samyoul) [reply src/status_im/ui/screens/wallet/account/views.cljs]: Reverted

**Comment by Serhy** [comment]
@cammellos with the latest build https://status-im-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-210208-101448-41b13e-pr-universal.apk on Android 11 it's not crashing when I open Dapp, however it crashes when I close this DApp after usage by tapping `X` icon:

Full device crash log: 
[crash_when_closing_dapp.txt](https://github.com/status-im/status-react/files/5943191/crash_when_closing_dapp.txt)

**Comment by Serhy** [comment]
Looks good to me on Android 11, Android 9, iOS 14, - modal view of list of services as well as each webview opens in modal form. Tested in dark and light mode.

**Pull request merged**